lib/src/clojure_lsp/shared.clj

```diff
@@ -171,7 +171,7 @@
         (uri-obj->filepath uri-obj)))))
 
 (defn ensure-jarfile [uri db]
-  (let [jar-scheme? (= "jar" (get @db [:settings :dependency-scheme]))]
```

Comment:
As with the `queries` namespace, the `shared` namespace now deals exclusively with a derefed db, not an atom. Again, this reveals many places that look like they might be updating the db but aren't.

Reply:
sounds perfect too!